dbms/src/Storages/KVStore/MultiRaft/RegionData.cpp

```diff
@@ -131,22 +109,23 @@ RegionData::WriteCFIter RegionData::removeDataByWriteIt(const WriteCFIter & writ
     std::ignore = value;
     std::ignore = key;
 
+    RegionDataMemDiff delta;
+
     if (decoded_val.write_type == RecordKVFormat::CFModifyFlag::PutFlag)
     {
         auto & map = default_cf.getDataMut();
 
         if (auto data_it = map.find({pk, decoded_val.prewrite_ts}); data_it != map.end())
         {
-            auto delta = RegionDefaultCFData::calcTiKVKeyValueSize(data_it->second);
-            cf_data_size -= delta;
+            delta.sub(RegionDefaultCFData::calcTotalKVSize(data_it->second));
             map.erase(data_it);
-            reportDealloc(delta);
         }
     }
 
-    auto delta = RegionWriteCFData::calcTiKVKeyValueSize(write_it->second);
-    cf_data_size -= delta;
-    reportDealloc(delta);
+    delta.sub(RegionWriteCFData::calcTotalKVSize(write_it->second));
+    cf_data_size += delta.payload;
+    decoded_data_size += delta.decoded;
+    recordMemChange(delta);
```

Comment:
```suggestion
    recordMemChange(delta);
```
cf_data_size and decoded_data_size has been changed in `recordMemChange(delta)`

Reply:
I have tried your `recordMemChange`, and now I think we'd better not combined the `root_of_kvstore_mem_trackers` with `cf_data_size += delta.payload;`. A change in memory in a certain region doesn't necessarily change the total memory usage. IMO it doesn't make the code clearer. We still have to handle the `cf_data_size` anywhere else.